### PR TITLE
Sign System.JSON with original Silverlight key

### LINKED ISSUE #18172: System.Json assembly identity
We added System.Json with https://github.com/dotnet/corefx/pull/9897 and later updated the version number to match mono with https://github.com/dotnet/corefx/issues/9938. However we still haven't taken into account the public key. Currently it has the MSFT public key (soon to be the Open key https://github.com/dotnet/corefx/pull/18165) and so any existing binary references will still not bind to this assembly. 

We need to decide if we want this binary to exactly match the one we shipped with SL and if so we should give it the same complete assembly identity or we should rename it to a new assembly name and have it fall inline with any other new assembly in corefx.

I think I would make it match the exact assembly identity.

cc @stephentoub @terrajobst opinions? 

@karelz @danmosemsft do you know who owns this library?

### PR DESCRIPTION
**Pull request by tarekgh**
Fixes #18172

### CODE REVIEW

**Comment by tarekgh** [comment]
@weshaggard PTAL

**Comment by tarekgh** [comment]
https://github.com/dotnet/buildtools/pull/1440

File: src/System.Json/src/System.Json.csproj [hunk 1]
```diff
@@ -3,6 +3,7 @@
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   <PropertyGroup>
     <ProjectGuid>{B11B5B76-F583-44EA-A4B7-3ABD990594F6}</ProjectGuid>
+    <AssemblyKey>SilverlightExtension</AssemblyKey>
```
Comment (danmosemsft) [inline src/System.Json/src/System.Json.csproj]: Maybe a note why we're picking this key?

File: src/System.Json/src/System.Json.csproj [hunk 2]
```diff
@@ -3,6 +3,10 @@
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   <PropertyGroup>
     <ProjectGuid>{B11B5B76-F583-44EA-A4B7-3ABD990594F6}</ProjectGuid>
+    <!-- 
+        System.Json mainly provide compatibility for Mono apps. we should keep the identity of this library matching
```
Comment (weshaggard) [inline src/System.Json/src/System.Json.csproj]: It isn't just about Mono. I would say this is a library for compat purposes so we want to keep the same identity as it originally shipped with in SL.

**Pull request merged**